From b8f0b78b055941e1512f8967b8d40236be9a78af Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Wed, 3 Jun 2020 06:00:04 +0530
Subject: [PATCH] gsocket: if_nametoindex is not available on WinRT

if_indextoname() is also not available, and since that's the only way
to get interface names on Windows, we can simply disable this code.
---
 gio/gsocket.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gio/gsocket.c b/gio/gsocket.c
index bd5549f..6348c63 100644
--- a/gio/gsocket.c
+++ b/gio/gsocket.c
@@ -2259,7 +2259,11 @@ g_socket_w32_get_adapter_ipv4_addr (const gchar *name_or_ip)
    *  function if_nametoindex will return >=1 if a valid index, or 0=no match
    *  valid index will be used later in GetAdaptersAddress loop for lookup of adapter IP address
    */
+#ifdef G_WINAPI_ONLY_APP
+  if_index = 0;
+#else
   if_index = if_nametoindex (name_or_ip);
+#endif
 
   /* Step 3: Prepare wchar string for friendly name comparison */
   if (if_index == 0)
-- 
2.38.1

